rfd/0138/README.md

```diff
@@ -0,0 +1,146 @@
+```
+{
+    "description": "SDC zones: allow all TCP from admin_rack99 net",
+    "enabled": true,
+    "owner_uuid": "930896af-bf8c-48d4-885c-6573a94b1853",
+    "rule": "FROM subnet 10.222.222.0/24 TO tag \"smartdc_role\" ALLOW tcp PORT all"
+}
+```
+* Each admin subnet must have a gateway defined in its NAPI network properties. 
+* Admin networks are automaticaly re-created when they go missing.  Therefore, multi-subnet admin networks need to be part of the initial DC deployment, otherwise any new admin network nics on the same subnet will conflict (DAD) with the new admin network nics.
+* Fabrics will need to be enabled `sdcadm post-setup fabrics` or booter's config will need to be modified to set `disableBootTimeFiles: false`.  Going forward we may make this the default.
+
+
+## Assumptions
+* A TOR switch with DHCP relay capability will be used to route traffic for each of the admin subnets. 
+* The "admin" network pool will only contain a single network per nictag.
+
+
+## Implementation
+One main issue is that most services and configuration scripts require at least one CN nic to have the nictag 'admin'
+
+The convention throughout SDC is to use the format: 
+'<tag>_nic=<mac addr>'
+to denote which nic should be assigned which nictag.
+
+Initially the approach was to simply override 'admin_xxx' with the admin network's properties (i.e. xxx == ip, mac, etc so that admin_nic=<mac address of admin_rack99_nic>).  Unfortunately that won't work because as noted above most of the configuration processing code use the actual property name as a tag name 'foo_nic' for 'foo' nictag.  When we run `nictagadm list` we will see:
+
+```
+NAME           MACADDRESS         LINK           TYPE            
+admin_rack99   d2:9c:54:2c:8c:a4  vioif0         normal          
+admin          d2:9c:54:2c:8c:a4  vioif0         normal   
+```
+
+So any agents looking for the CN's nictags will see a single nic tagged with two different tags.  This may lead other agents and services to believe the CN is on the "admin" subnet(NAPI network), when it is really only on the "admin_rack99" subnet(NAPI network).
+
+This leaves us with two possible approaches:
+a) modify the existing behavior of every script that parses 'admin_nic' as '<nictag>_nic'
+
+OR
+
+b) Add a new property "admin_tag" to denote an alternately tagged admin network (e.g. "admin_tag=admin_rack99".  Then in each script that parses the usbkey or networking.json config we could have the value of "admin_tag" override the string "admin" in the results.  Said another way, if "admin_tag" is set, any configuration file consumers will essentially ignore values of the form "admin_XXX", unless "admin_tag=admin" (which would be the default behavior, and there would be no use in actually setting that...).  So if "admin_tag=admin_rack99" then the configuration file consumers would use values of "admin_rack99" instead of those from "admin".  `nictagadm` would simply skip over anything of the form "admin_XXX".
```

Comment:
As far as Triton and Manta GZ agents go, they can also switch to using `auto.ADMIN_IP` now that [TRITON-298](https://smartos.org/bugview/TRITON-298) is in.

Reply:
Will do.